config/jobs/kubernetes/sig-node/dynamic-resource-allocation.yaml

```diff
@@ -242,7 +242,7 @@ periodics:
         - --deployment=node
         - --env=KUBE_SSH_USER=core
         - --gcp-zone=us-west1-b
-        - '--node-test-args=--feature-gates="DynamicResourceAllocation=true" --service-feature-gates="AllAlpha=true,AllBeta=true" --runtime-config=api/alpha=true,api/beta=true --container-runtime-endpoint=unix:///var/run/crio/crio.sock --container-runtime-process-name=/usr/local/bin/crio --container-runtime-pid-file= --kubelet-flags="--cgroup-driver=systemd --cgroups-per-qos=true --cgroup-root=/ --runtime-cgroups=/system.slice/crio.service --kubelet-cgroups=/system.slice/kubelet.service" --extra-log="{\"name\": \"crio.log\", \"journalctl\": [\"-u\", \"crio\"]}"'
+        - '--node-test-args=--feature-gates="DynamicResourceAllocation=true,KubeletCrashLoopBackOffMax=true"--service-feature-gates="AllAlpha=true,AllBeta=true" --runtime-config=api/alpha=true,api/beta=true --container-runtime-endpoint=unix:///var/run/crio/crio.sock --container-runtime-process-name=/usr/local/bin/crio --container-runtime-pid-file= --kubelet-flags="--cgroup-driver=systemd --cgroups-per-qos=true --cgroup-root=/ --runtime-cgroups=/system.slice/crio.service --kubelet-cgroups=/system.slice/kubelet.service" --extra-log="{\"name\": \"crio.log\", \"journalctl\": [\"-u\", \"crio\"]}"'
```

Comment:
Would it be better to enable only service-feature-gates that are needed for DRA?

Reply:
This test seems to be designed to run with all alpha/beta gates enabled. We have another test where we only enable DRA.

Reply:
Thank you for the explanations! It makes sense to me now. Can you explain why we have to enable `KubeletCrashLoopBackOffMax` explicitly? I mean why kubelet fails to start without it? Should we change kubelet code instead?

Reply:
So in this job we are enabling all alpha and beta feature gates via -service-feature-gates. This does not that enables kubelet feature gates.

So what is going on is that kubelet has this gate disabled while the apiserver has it enabled. So I think we are getting an odd state and I don’t think it is a big in kubelet.

Reply:
It looks like kubelet config is generated using service feature gates. Do you happen to know where does it happen in e2e_node code?

Reply:
found it: https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/kubeletconfig/kubeletconfig.go#L147
It looks like e2e_node gets it from the service's /configz API.

Reply:
Reading the code, service feature gates are used for the apiserver while kubelet's feature gate option is used for Kubelet.

Services: https://github.com/kubernetes/kubernetes/blob/8fe10dc378b7cc3b077b83aef86622e1019302d5/test/e2e_node/e2e_node_suite_test.go#L148

Kubelet: https://github.com/kubernetes/kubernetes/blob/8fe10dc378b7cc3b077b83aef86622e1019302d5/test/e2e_node/services/kubelet.go#L286

Reply:
Even after going through this discussion thread it is not clear to me why `KubeletCrashLoopBackOffMax=true` is needed. It was only added here, but not the pre-submit jobs, and those work fine without it.

It should be possible to enable more features in the apiserver than in kubelet. That's a normal situation that occurs during the recommended update process: first update the apiserver, potentially enable new features, then update the kubelet.

I am going to step around this by removing the feature gate gates from all E2E node tests for DRA because we shouldn't need it anymore (all code in kubelet is beta), but it will come back when we have alpha features in the kubelet.